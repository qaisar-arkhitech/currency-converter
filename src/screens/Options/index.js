import React from "react"
import {SafeAreaView, StatusBar, ScrollView} from "react-native"
import PropTypes from "prop-types"
import Entypo from "react-native-vector-icons/Entypo"
import flow from "lodash/flow"
import themeStyles from "./styles"
import connect from "./connect"
import {RowItem} from "../../components/RowItem"

// Options Component content
export const Options = ({navigation, styleableTheme}) => {
  const styles = themeStyles(styleableTheme)
  return (
    <SafeAreaView style={styles.root}>
      <StatusBar barStyle="default" backgroundColor={styleableTheme[500]} />
      <ScrollView>
        <RowItem
          title="Themes"
          onPress={() => navigation.push("Themes")}
          rightIcon={<Entypo name="chevron-right" size={20} />}
        />
      </ScrollView>
    </SafeAreaView>
  )
}

// Options Proptypes
Options.propTypes = {
  navigation: PropTypes.object.isRequired,
  styleableTheme: PropTypes.object,
}

// Options Default props
Options.defaultProps = {
  styleableTheme: {},
}

export default flow([connect])(Options)
