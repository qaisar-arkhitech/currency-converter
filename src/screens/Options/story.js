import React from "react"
import Options from "."

export default {
  title: "Options",
  storyFn() {
    return (
      <div>
        <Options />
      </div>
    )
  },
}
