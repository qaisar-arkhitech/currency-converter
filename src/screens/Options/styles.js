import {StyleSheet} from "react-native"

// Options: Styling here
const styles = (styleableTheme) => {
  return StyleSheet.create({
    root: {
      flex: 1,
      backgroundColor: styleableTheme.pri50,
    },
  })
}

export default styles
