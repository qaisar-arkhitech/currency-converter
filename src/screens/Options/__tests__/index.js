import React from "react"
import renderer from "react-test-renderer"
import Options from ".."

const createProps = () => ({
  classes: {},
})

describe("<Options />", () => {
  it("Does not explode", () => {
    const props = createProps()
    const component = renderer.create(<Options {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
