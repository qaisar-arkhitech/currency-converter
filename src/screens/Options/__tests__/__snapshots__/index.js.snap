// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Options /> Does not explode 1`] = `
<View>
  <Text>
    Options
  </Text>
</View>
`;
